docs/tests_in_tilt.md

```diff
@@ -0,0 +1,109 @@
+</figure>
+
+Just like any other resource managed by Tilt, you can drill into a test and see logs and other details.
+
+In the log view, you can filter the sidebar by item type (if, say, you want to see only tests, or you want to filter tests out and focus on your app in the cluster). You can also toggle "Alerts On Top" sorting, to make it even more visible when something has gone wrong and needs your attention
+
+<figure>
+    <img src="/assets/img/tests-in-tilt/sidebar-alerts-on-top.png" alt="sidebar displaying only tests, with alerts sorted to the top">
+    <figcaption>Here's an example of sidebar filtering (displaying only tests). In "Alerts on Top" mode, any cards with alerts are sorted to the top of the sidebar for easy visibility</figcaption>
+</figure>
+
+### When Tests Run
+Just as with a `local_resource`, you can associate files with a test via the `deps=[...]` parameter. When any of these file dependencies change, Tilt will re-execute the test:
+
+```python
+# A test depending on an entire directory
+test('cart-tests', 'go test ./pkg/cart', deps=['./pkg/cart'])
+
+# A test depending on specific relevant files
+test('unit-test-remittance', 'pytest remittance_test.py',
+     deps=['app/remittance.py', 'app/remittance_test.py', 'app/payment_utils.py'])
+```
+(More on how to intelligently set your dependencies below.)
+
+Sometimes, you don't want your test(s) to execute automatically. Maybe you want to turn off automatic-ness for all but a few tests you're iterating on, or prevent an expensive integration test from running all the time. Use the "auto" trigger in the UI to toggle automatic execution on or off for a given test.
+
+<figure>
+    <img src="/assets/img/tests-in-tilt/auto-toggles" alt="a close-up of several cards in the sidebar, with 'auto' variously toggled on or off">
+    <figcaption>In this example, tests "foo" and "beep" are in auto mode--they will execute automatically when any of their `deps` are changed.</figcaption>
+</figure>
+
+You can also set default AUTO values for your tests via the Tiltfile, using [the `trigger_mode` parameter](https://docs.tilt.dev/manual_update_control.html).
+
+### Programmatically Registering Tests
+Part of what's great about Starlark (the dialect of Python that Tiltfiles are written in) is that it's a _programming language_, so you can use it for complex stuff. In particular, you can use it to programmatically register your tests to Tilt in whichever way works best for you. Here are some examples:
+
+#### Your entire unit test suite
+If your whole unit test suite runs relatively quickly/has caching (like Go tests), you can just run the whole thing when relevant files change:
+```python
+def all_go_files(path):
+  return str(local('find . -type f -name "*.go"')).split("\n")
+
+test('go-tests', 'go test ./... -timeout 30s', deps=all_go_files())
+```
+
+#### Go tests by package
+This approach gives you a little more granularity:
```

Comment:
ELI5 for the kids at the back maybe? "This way your tests will be split into different cards" or something?

Reply:
+1

also hit on the change in test run behavior, like "Each package's tests will only run when that package changes"

NB this also has a key distinction from the previous approach: `go test` does have good caching, but a test's results will be invalidated when one of its dependencies change. If package `foo` depends on `bar`. With the first approach, when `bar` changes, we re-rerun both `foo` and `bar`'s tests, and neither is cached. With the second approach, when `bar` changes, we do not re-run `foo`'s tests, since we didn't add `bar` to its deps. This speeds things up but also potentially misses breaks.